### PR TITLE
[Fix] fix a bug of showing kmeans losses

### PR DESCRIPTION
**Pull request by Kaminyou**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
To fix the bug that occurs during showing k-means losses in the training processing.
```
TypeError: unsupported format string passed to numpy.ndarray.__format__
```

## Modification
```python
if verbose:
    print(f'k-means loss evolution: {losses:0}')
```
to 
```python
if verbose:
    print(f'k-means loss evolution: {losses}')
```
As `losses` is a `numpy::ndarray`.

## BC-breaking (Optional)

None

## Use cases (Optional)

None

## Checklist

**Before PR**:

- [v] Pre-commit or other linting tools are used to fix the potential lint issues.
- [v] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [v] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [v] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [v] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [v] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Review by Jiahao000 — COMMENTED** [review]
Could you fix the unit test to pass the check?

**Comment by Kaminyou** [comment]
@Jiahao000 Sorry for the delay but currently I can hardly find out the problem in my modified `tests/test_utils/test_clustering.py` which cannot pass the `codecov/patch`.

**Comment by Kaminyou** [comment]
@Jiahao000 It seems that my modified `tests/test_utils/test_clustering.py` is not tracked by `codecov`. Could someone authorized add this change to the current unit test in `codecov`?

**Comment by fangyixiao18** [comment]
I think the reason might be the `pytest.mark.skipif` decorator in `test_clustering.py`, as the original code needs GPU to run.

**Comment by Kaminyou** [comment]
@fangyixiao18 I remove the GPU restrict and see it would also lead to errors during the test. Besides, `faiss` generally requires GPU resources. What should I do to bypass all these problems?

**Comment by fangyixiao18** [comment]
> @fangyixiao18 I remove the GPU restrict and see it would also lead to errors during the test. Besides, `faiss` generally requires GPU resources. What should I do to bypass all these problems?

Currently, the CI doesn't support GPU so I wrote the `skipif` decorator.  
I think the `unittest.mock` would help to run the clustering unittest without GPU, would you like to help us to revise the unittest if you have time?

**Comment by Kaminyou** [comment]
@fangyixiao18 I am not sure how to solve this problem even with `unittest.mock` without modifying multiple lines in `utils/clustering.py`. However, if I cluster those function calls related to `faiss` and use `unitest.mock` to bypass GPU usage, would `codecov` still show `Added line #Lxxx was not covered by tests`?

**Comment by fangyixiao18** [comment]
@Kaminyou Thanks a lot！Could you please run the pre-commit and check the codes to pass the lint check？

**Comment by Kaminyou** [comment]
@fangyixiao18 Everything has been done. My coworker found a bug in `isort config` which make my code not able to pass the lint check. Please first deal with that [PR](https://github.com/open-mmlab/mmselfsup/pull/184) and I will rebase and push again.

**Comment by Kaminyou** [comment]
@fangyixiao18 @Jiahao000 Please check for my update.

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import pytest
+from unittest.mock import patch
+
+
+@pytest.fixture(scope='session')
```
Comment (fangyixiao18) [inline tests/conftest.py]: can this function be moved to `test_clustering.py`? I think it is no need to be a 'session' scope function.

**Review by fangyixiao18 — APPROVED** [review]

**Review by Jiahao000 — APPROVED** [review]
@Kaminyou All checks have passed. Thanks a lot!

**Pull request merged**